src/AssertionInterface.php

```diff
@@ -1,21 +1,18 @@
 <?php
 /**
- * Zend Framework (http://framework.zend.com/)
- *
- * @link      http://github.com/zendframework/zf2 for the canonical source repository
- * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
- * @license   http://framework.zend.com/license/new-bsd New BSD License
+ * @see       https://github.com/zendframework/zend-permissions-rbac for the canonical source repository
+ * @copyright Copyright (c) 2005-2018 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-permissions-rbac/blob/master/LICENSE.md New BSD License
  */
 
+declare(strict_types=1);
+
 namespace Zend\Permissions\Rbac;
 
 interface AssertionInterface
 {
     /**
      * Assertion method - must return a boolean.
-     *
-     * @param  Rbac    $rbac
-     * @return bool
      */
-    public function assert(Rbac $rbac);
+    public function assert(Rbac $rbac, RoleInterface $role = null, string $permission = null) : bool;
```

Comment:
I'm still not entirely convinced these arguments should be optional.

While I understand the argument of keeping BC breaks to a minimum, I also think we could address this in a new v2.X minor release by doing the following:

- Updating the `Rbac` class to pass the role and permission when calling an assertion.
- Documenting in migration notes to users that they should add these arguments to their assertions in order to prepare for the v3 release. Their assertions can ignore them for now — or even update their logic to make use of them!

Taking those steps allows the v3 release to make these required, as they are always passed.

Reply:
@weierophinney ok, this plan sounds good to me.

Reply:
In talking with @ezimuel today, we realized that for anybody implementing the `AssertionInterface`, they will need to make a signature change when upgrading to version 3 regardless of any changes we'd make in the v2 release branch. As such, we will not do another v2 release, and move ahead instead with the v3 release.